packages/mux-active-viewer-count/src/index.ts

```diff
@@ -0,0 +1,225 @@
+import { globalThis } from 'shared-polyfills';
+
+export const subscribeViewerCount = (
+  token: string,
+  pollInterval: number,
+  callback: (views: number) => void,
+  errorCb: (errorMsg: string) => void
+) => {
+  const url = `https://stats.mux.com/counts?token=${token}`;
+  const controller = new AbortController();
+  const signal = controller.signal;
+  let timeoutId: number | undefined;
+  let aborted = false;
+  const fetchViewerCountPoll: () => Promise<any> = async () => {
+    // If the polling has been aborted (via an "unsubscribe()"),
+    // we can simply bail on the recursion.
+    if (aborted) return Promise.resolve();
+    // GET the latest view count, providing an abort signal
+    // for unsubscription.
+    return (
+      fetch(url, { signal })
+        // Grab the JSON value of the response
+        .then((resp) => resp.json())
+        // Confirm that response wasn't an error and the JSON
+        // has the expected data
+        .then((respObj) => {
+          const views = respObj?.data?.[0]?.views;
+          if (!!respObj?.error || views == null) {
+            // If not, treat as an error.
+            return Promise.reject(respObj?.error ?? 'no data in response');
+          }
+          // Otherwise, we successfully retrieved the latest views, so
+          // provide that info out via `callback()`.
+          callback(views);
+          return views;
+        })
+        // Catch and invoke errorCb before timeout + re-fetch. This allows
+        // for re-fetching by default, but provides the opportunity
+        // to unsubscribe externally via `errorCb` if desired (CJP)
+        .catch(errorCb)
+        // Wait the duration of the polling interval before restarting
+        // the next fetch
+        .then(() => {
+          return new Promise((resolve) => {
+            timeoutId = setTimeout(async () => {
+              resolve(undefined);
+            }, pollInterval * 1000);
```

Comment:
here too `async` is unneeded.

could be shortened to

```suggestion
            timeoutId = setTimeout(resolve, pollInterval * 1000);
```

Reply:
Fixed!